qa/rpc-tests/p2p-segwit.py

```diff
@@ -57,6 +57,7 @@ def on_block(self, conn, message):
         self.last_block.calc_sha256()
 
     def on_getdata(self, conn, message):
+        self.last_last_getdata = self.last_getdata
```

Comment:
This isn't quite right -- while storing the second-to-last `getdata` request would be sufficient for testing whether the block is requested via direct fetch (where a getdata request goes out immediately), it would be insufficient for testing whether a `getdata` is going out via the parallel download mechanism, because the `getdata` for the block and the `getdata` for the transaction you're announcing would get bundled into a single message.

Perhaps the best approach here would be to keep a map that tracks block getdata requests by hash which can be cleared in between tests?